Meadowlark-js/src/repository/MongoEntityRepository.ts

```diff
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+import { Collection, Db, MongoClient, WithId } from 'mongodb';
+import { ForeignKeyItem } from '../model/ForeignKeyItem';
+import { EntityInfo } from '../model/EntityInfo';
+import { Security } from '../model/Security';
+import { ValidationOptions } from '../model/ValidationOptions';
+import { DeleteResult, GetResult, OwnershipResult, PutResult } from './BaseDynamoRepository';
+
+interface Entity {
+  // A string hash of the project name, entity type, entity version and body of
+  // the API document. This is identical to PK identifier in the DynamoDB
+  // implementation. This field will be a unique index on the collection.
+  id: string;
+
+  // The MetaEd project name the entity is defined in e.g. "EdFi" for a data standard entity.
+  project_name: String;
+
+  // The entity type as a string e.g. "Student".
+  entity_type: String;
+
+  // The entity version as a string. This is the same as MetaEd project version
+  // the entity is defined in e.g. "3.3.1-b" for a 3.3b data standard entity.
+  entity_version: String;
+
+  // The ODS/API document body as a sub-document.
+  api_doc: any;
+
+  // An array of _ids extracted from the ODS/API document body for all externally
+  // referenced documents.
+  out_refs: string[];
+}
+
+async function naiveGetEntities(): Promise<Collection<Entity>> {
+  const client = new MongoClient('mongodb://localhost:27017');
+
+  await client.connect();
+  const db: Db = client.db('meadowlark');
+  const entities: Collection<Entity> = db.collection('entities');
+
+  // Note this will trigger a time-consuming index build if the indexes do not already exist.
```

Comment:
I wonder if we can pre-create these indexes in a full implementation?

Maybe just uploading (and then retrieving) descriptors would be sufficient?